bin/packages/watch.js

```diff
@@ -21,6 +21,12 @@ const exists = ( filename ) => {
 	} catch ( e ) {}
 	return false;
 };
+
+// Exclude deceitful source-like files, such as editor swap files.
+const isSourceFile = ( filename ) => {
+	return filename.indexOf( '.' ) !== 0 && filename.match( /\.js$/ );
```

Comment:
Minor: Generally recommend using `RegExp#test` if you only care about boolean value of match, as there can tend to be a performance difference:

https://jsperf.com/test-vs-match-regex

Also wonder if this could be simplified to just the regular expression, skipping the `String#indexOf`:

```js
return /.\.js$/.test( filename );
```

But I'm wondering: What are we excluding with that condition? Isn't this a full URL path? This isn't entirely clear by the comment.

Reply:
I agree with both: I was thinking I should use `test`, but `match` just stuck; targeting a leading `.` was my first reflex to exclude "hidden files", but I think we're alright with just `/.\.js$/`.

Reply:
Amended in 2559cff